aesara/tensor/subtensor_opt.py

```diff
@@ -1666,3 +1668,98 @@ def local_subtensor_SpecifyShape_lift(fgraph, node):
         new_shape_arg = as_tensor(shape_arg[indices], dtype=np.int64, ndim=1)
 
     return [specify_shape(new_obj_arg, new_shape_arg)]
+
+
+def equal_idxs(idx1, idx2):
+    if isinstance(idx1, slice) and isinstance(idx2, slice):
+        return (
+            equal_computations([idx1.start], [idx2.start])
+            and equal_computations([idx1.stop], [idx2.stop])
+            and equal_computations([idx1.step], [idx2.step])
+        )
+    else:
+        return equal_computations([idx1], [idx2])
+
+
+@register_specialize
+@local_optimizer([Join])
+def local_join_subtensors(fgraph, node):
+    r"""Simplify contiguous :clase:`Subtensor`s inside a :class:`Join`.
+
+    join((x[:3], x[3:5]), axis=0) -> x[:5]
```

Comment:
```suggestion
        join((x[:3], x[3:5]), axis=0) -> x[:5]
```

Reply:
pre-commit disagrees with that extra indentation

Reply:
The indent in the docstring?  Weird.

Reply:
pre-commit just overrides it. Let me see which hook is it

Reply:
Well, perhaps we can just add a `..code:` above it and get some formatting, or wrap it in backticks.

Reply:
Something like this?
```python
r"""Simplify contiguous :class:`Subtensor`\s inside a :class:`Join`.

..code:
  join((x[:3], x[3:5]), axis=0) -> x[:5]
"""
```

Reply:
I think it's `.. code::` (or `.. code:: python`).